From 397872759e01324db6ced2fd3fbc7767b807d5dd Mon Sep 17 00:00:00 2001
From: Adam Ford <aford173@gmail.com>
Date: Mon, 11 May 2020 09:39:16 -0500
Subject: [PATCH 30/31] arm64: dts: beacon-imx8mm-spidev: Fix CS pins on Beta
 hardware

Per the SPI bindings documentation from kernel 4.19.35:
  While the native chip select lines can be used, they appear to always
  generate a pulse between each word of a transfer. Most use cases will
  require GPIO based chip selects to generate a valid transaction.

Using the native CS pin fails on beta hardware which is likely related to
the above note. This patch changes the chip select to GPIO and routes
to gpio5-13 on the Beta hardware for spidev devices.

Signed-off-by: Adam Ford <aford173@gmail.com>
---
 arch/arm64/boot/dts/freescale/beacon-imx8mm-spidev.dts | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/beacon-imx8mm-spidev.dts b/arch/arm64/boot/dts/freescale/beacon-imx8mm-spidev.dts
index 46751be0ac4f..1865b242c325 100644
--- a/arch/arm64/boot/dts/freescale/beacon-imx8mm-spidev.dts
+++ b/arch/arm64/boot/dts/freescale/beacon-imx8mm-spidev.dts
@@ -30,7 +30,8 @@
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_ecspi1>;
 	status = "okay";
-	cs-gpios = <&gpio5 9 0>;
+	cs-gpios = <&gpio5 9 GPIO_ACTIVE_LOW>;
+
 	spidev1@0x00 {
                compatible = "spidev";
                spi-max-frequency = <5000000>;
@@ -39,8 +40,6 @@
 };
 
 &ecspi2 {
-	cs-gpios = <&gpio5 13 0>;
-	
 	/delete-node/ at25@0;
 	
 	spidev1@0x00 {
@@ -56,7 +55,7 @@
 			MX8MM_IOMUXC_ECSPI1_SCLK_ECSPI1_SCLK		0x82
 			MX8MM_IOMUXC_ECSPI1_MOSI_ECSPI1_MOSI		0x82
 			MX8MM_IOMUXC_ECSPI1_MISO_ECSPI1_MISO		0x82
-			MX8MM_IOMUXC_ECSPI1_SS0_ECSPI1_SS0		0x82
+			MX8MM_IOMUXC_ECSPI1_SS0_GPIO5_IO9		0x41
 		>;
 	};
 };
-- 
2.17.1

